src/wallet/wallet_tx_builder.cpp

```diff
@@ -32,38 +32,34 @@ CalcZIP317Fee(
     std::vector<CTxOut> vout{};
     size_t sproutOutputCount{}, saplingOutputCount{}, orchardOutputCount{};
     for (const auto& payment : payments) {
-        std::visit(match {
+        examine(
+            payment.address,
             [&](const CKeyID& addr) {
                 vout.emplace_back(payment.amount, GetScriptForDestination(addr));
             },
             [&](const CScriptID& addr) {
                 vout.emplace_back(payment.amount, GetScriptForDestination(addr));
             },
-            [&](const libzcash::SaplingPaymentAddress&) {
-                ++saplingOutputCount;
-            },
-            [&](const libzcash::OrchardRawAddress&) {
-                ++orchardOutputCount;
-            }
-        }, payment.address);
+            [&](const libzcash::SaplingPaymentAddress&) { ++saplingOutputCount; },
+            [&](const libzcash::OrchardRawAddress&) { ++orchardOutputCount; });
```

Comment:
```suggestion
            [&](const libzcash::OrchardRawAddress&) { ++orchardOutputCount; }
        );
```
and similarly for the other uses.

Reply:
It sounds like this was discussed in Ancient Issues today, so maybe we’re all on the same page already, but I wanted to check before I made all the changes. So, here’s a bit of summary, mostly for @nuttycom, @ebfull, and @str4d to respond to (and for @daira to call me out if I misrepresent something):

First, we can either get clang-format to approximate what we want, and get autoformatting, or add `match` (née `examine`) to `WhitespaceSensitiveMacros` (WSM) which effectively treats every call as if it were wrapped in `clang-format off`.

There are other `*Macros` lists (like [`ForEachMacros`](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#foreachmacros)) that are less all-or-nothing than WSM, but I don’t think we can massage `match` into any of them.

Personally, I’m not that keen on WSM, especially for `match`es that don’t consist of 1-lineish lambdas – I like things being formatted for me. But, I’d much rather give up those sections of code formatting than lose them all. I also don’t really like the `)` being treated like a brace in this one case, especially without the trailing `,`. So, my own preferences are

1. let clang-format do what it wants
    ```c++
    match(
        meh,
        [](const Foo& foo) { … },
        [](const Bar& bar) { … });
    ```
2. figure out how to make a block macro for this (which I don’t think can work with the structure of this operation and CPP, but I’m happy to be proven wrong), like
    ```c++
    match(meh) {
        [](const Foo& foo) { … },
        [](const Bar& bar) { … },
    }
    ```
3. manually do what we want here, using WSM
    ```c++
    match(meh,
        [](const Foo& foo) { … },
        [](const Bar& bar) { … }
    );
    ```

But, none of that is very strong. I just want some consensus before I make the edits.